packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -80,12 +93,7 @@ export abstract class AbstractAsyncHooksContextManager
       writable: false,
       value: target.length,
     });
-    /**
-     * It isn't possible to tell Typescript that contextWrapper is the same as T
-     * so we forced to cast as any here.
-     */
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    return contextWrapper as any;
+    return contextWrapper as T;
```

Comment:
We didn't have to cast to `any` here, since the `contextWrapper` satisfies the call signature of `T` (because I typed all the arguments and return type in accordance to the original `T`).

However, a cast is necessary here (and technically, unsafe) because `T` can encompass more than just the call signature. For example, if the input function has a custom property on it (`fn.foo = "foo fn"`), then it will be lost with this wrapper. Probably not an issue in the real world in this case, but TypeScript isn't wrong to flag it.

And, in general, even if we _have_ to do a completely unrelated unsafe cast, casting into `unknown` and then to the target type would always work, without having to involve `any` and disabling the lint rule.